src/main.ts

```diff
@@ -0,0 +1,628 @@
+  }
+
+  const bytes = pipe(
+    refresh_token,
+    (x) => Buffer.from(x, "base64"),
+    (buffer) =>
+      new Uint8Array(buffer.buffer, buffer.byteOffset, buffer.byteLength)
+  );
+  const refreshToken = new TextDecoder().decode(
+    await decrypt(bytes, config.key)
+  );
+
+  let accessToken = "";
+  let accessTokenExpiration: string = new Date().toISOString();
+  const token = accessTokenMap.get(userID);
+  if (token && token.expiresEpoch > Date.now()) {
+    accessToken = token.accessToken;
+    accessTokenExpiration = new Date(token.expiresEpoch).toISOString();
+  }
+
+  const Client = getGitHubClient();
+  const client = new Client({
+    authStrategy: createOAuthUserAuth,
+    auth: {
+      clientType: "github-app",
+      clientId: config.oauth.github.clientID,
+      clientSecret: config.oauth.github.secret,
+      token: accessToken,
+      refreshToken: refreshToken,
+      expiresAt: accessTokenExpiration,
+      refreshTokenExpiresAt: new Date(expiration_date).toISOString(),
+    } satisfies GitHubAppStrategyOptionsExistingAuthenticationWithExpiration,
+  });
+
+  const [start, end] = await getDates();
+
+  // NEEDS TO BE SET IN PHILL
+  const repoName = "alumni-api";
+  const message: ListMessage = {
+    repo: repoName,
+    since: start,
+    until: end,
+    author: rows[0].user_name,
+    owner: "Panenco",
+  };
+
+  const listPrs = await listPullRequests(client, message);
+  const listCommits = await paginatedCommits(client, message);
+
+  return c.text(await summarize(listPrs, listCommits, repoName));
+});
+
+interface TokenRow {
+  user_id: string;
+  refresh_token: string;
+  expiration_date: number;
+  user_name: string;
+}
+
+/*
+Utility function for LSP to kick-in
+ */
+function graphql(query: string) {
+  return query;
+}
+
+type ListMessage = {
+  repo: string;
+  since: string;
+  until: string;
+  author: string;
+  owner: string;
+};
+
+async function listPullRequests(
+  client: Octokit,
+  message: ListMessage
+): Promise<string> {
+  const query = graphql(`
+		query paginate($cursor: String) {
+			search(query: "is:pr author:${message.author} created:${message.since}..${message.until} repo:${message.owner}/${message.repo}",  type: ISSUE, first: 100, after: $cursor) {
+				nodes {
+					... on PullRequest {
+						title
+						state
+						url
+						createdAt
+            body
+            author{
+                login
+                }
+						repository {
+							nameWithOwner
+						}  
+					}
+				}
+				pageInfo {
+					endCursor
+					hasNextPage
+				}
+			}
+		}
+	`);
+
+  const iterator = client.graphql.paginate.iterator<{
+    search: SearchResultItemConnection;
+  }>(query);
+
+  const prs: PullRequest[] = [];
+  // iterate over pages
+  for await (const response of iterator) {
+    const { nodes } = response.search;
+    if (!nodes || nodes.length === 0) break;
+
+    prs.push(...(nodes as PullRequest[]));
+  }
+
+  let listPrs = prs.map((pr) => pr.body).join(", ");
```

Comment:
it should not be here

listPullRequests means that it will return a list (array) of pull requests, not a string

Reply:
If I rename the function getPullRequests is it ok? Since I only need to use the actual string I would prefer to do it like this. But I can change it if you want.